File: app/src/main/java/com/bitflaker/lucidsourcekit/database/daos/DreamClarityDao.java
package com.bitflaker.lucidsourcekit.database.daos;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.bitflaker.lucidsourcekit.database.entities.DreamClarity;

import java.util.List;

import io.reactivex.rxjava3.core.Completable;
import io.reactivex.rxjava3.core.Single;

@Dao
public interface DreamClarityDao {
    @Query("SELECT * FROM DreamClarity")
    Single<List<DreamClarity>> getAll();

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    Completable insertAll(DreamClarity... dreamClarities);

    @Delete
    Completable delete(DreamClarity dreamClarity);
}

File: app/src/main/java/com/bitflaker/lucidsourcekit/database/dreamjournal/daos/DreamClarityDao.kt
package com.bitflaker.lucidsourcekit.database.dreamjournal.daos

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.bitflaker.lucidsourcekit.database.dreamjournal.entities.DreamClarity

@Dao
interface DreamClarityDao {
    @Query("SELECT * FROM DreamClarity")
    suspend fun getAll(): MutableList<DreamClarity>

    @Insert(onConflict = OnConflictStrategy.Companion.IGNORE)
    suspend fun insertAll(dreamClarities: Array<DreamClarity>)

    @Delete
    suspend fun delete(dreamClarity: DreamClarity)
}
